src/callbacks_step/save_solution.jl

```diff
@@ -52,25 +63,64 @@ function Base.show(io::IO, ::MIME"text/plain", cb::DiscreteCallback{<:Any, <:Sav
   end
 end
 
+function Base.show(io::IO, ::MIME"text/plain",
+                   cb::DiscreteCallback{<:Any, <:PeriodicCallbackAffect{<:SaveSolutionCallback}})
+  @nospecialize cb # reduce precompilation time
 
-function SaveSolutionCallback(; interval=0,
+  if get(io, :compact, false)
+    show(io, cb)
+  else
+    save_solution_callback = cb.affect!.affect!
+
+    setup = [
+             "dt" => save_solution_callback.interval,
+             "solution variables" => save_solution_callback.solution_variables,
+             "save initial solution" => save_solution_callback.save_initial_solution ? "yes" : "no",
+             "save final solution" => save_solution_callback.save_final_solution ? "yes" : "no",
+             "output directory" => abspath(normpath(save_solution_callback.output_directory)),
+            ]
+    summary_box(io, "SaveSolutionCallback", setup)
+  end
+end
+
+
+function SaveSolutionCallback(; interval::Integer=0,
+                                dt=0.0,
                                 save_initial_solution=true,
                                 save_final_solution=true,
                                 output_directory="out",
                                 solution_variables=cons2prim)
 
-  solution_callback = SaveSolutionCallback(interval, save_initial_solution, save_final_solution,
+  if dt > 0
+    interval = dt
+  end
+
+  solution_callback = SaveSolutionCallback(interval,
+                                           save_initial_solution, save_final_solution,
                                            output_directory, solution_variables)
 
-  DiscreteCallback(solution_callback, solution_callback, # the first one is the condition, the second the affect!
-                   save_positions=(false,false),
-                   initialize=initialize!)
+  if dt > 0
```

Comment:
It is, but your suggestion above (using `nothing`) should fix it.

Reply:
Ah, right! Thanks.

What about the fact that the call to `SaveSolutionBack` itself will not be type stable anymore, since we create either a periodic or a discrete callback? Do you see a problem with that?

Reply:
With your proposed change, it will depend on the type of the keyword arguments, so it should be fine - but it would be good to verify this manually (on Julia v1.8 and v1.9).